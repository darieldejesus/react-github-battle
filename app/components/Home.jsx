import React, { Component } from 'react';
import { Link } from 'react-router-dom';

/**
 * Home page which only has a button which
 * anchors to the Battle section.
 */
class Home extends Component {
  render() {
    return (
      <div className='home-container'>
        <h1>Github Battle: Battle your friends...and stuff.</h1>
        <Link to='/battle' className='button'>
          Battle
        </Link>
      </div>
    )
  }
}

export default Home;
