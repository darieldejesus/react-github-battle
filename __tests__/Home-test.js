import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Home from '../app/components/Home';

describe('<Home /> (Snapshot)', () => {
  it('Home component renders a title and a Link', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
