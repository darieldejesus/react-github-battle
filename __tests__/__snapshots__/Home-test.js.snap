// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> (Snapshot) Home component renders a title and a Link 1`] = `
<div
  className="home-container"
>
  <h1>
    Github Battle: Battle your friends...and stuff.
  </h1>
  <a
    className="button"
    href="/battle"
    onClick={[Function]}
  >
    Battle
  </a>
</div>
`;
